src/conn/conn_cache.c

```diff
@@ -46,6 +46,120 @@ __cache_config_abs_to_pct(
     return (0);
 }
 
+/*
+ * __validate_cache_config --
+ *     Validate trigger and target values of given configs.
+ */
+static int
+__validate_cache_config(WT_SESSION_IMPL *session, const char *cfg[], bool create)
+{
+    WT_CACHE *cache;
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    bool debug_config;
+
+    conn = S2C(session);
+    cache = conn->cache;
+
+    WT_RET(__wt_config_gets(session, cfg, "debug_mode.configuration", &cval));
+    debug_config = cval.val;
+
+    /* The target size must be lower than the trigger size or we will never get any work done. */
+    if (create) {
+        if (cache->eviction_target >= cache->eviction_trigger)
+            WT_RET_MSG(session, EINVAL, "eviction_target must be lower than the eviction_trigger");
+        if (cache->eviction_dirty_target >= cache->eviction_dirty_trigger)
+            WT_RET_MSG(session, EINVAL,
+              "eviction_dirty_target must be lower than the eviction_dirty_trigger");
+        if (cache->eviction_updates_target >= cache->eviction_updates_trigger)
+            WT_RET_MSG(session, EINVAL,
+              "eviction_updates_target must be lower than the eviction_updates_trigger");
+        return (0);
+    }
+
+    /* Resetting values for reconfiguration. */
+    WT_RET(__wt_config_gets(session, cfg, "eviction_target", &cval));
+    cache->eviction_target = (double)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_trigger", &cval));
+    cache->eviction_trigger = (double)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_dirty_target", &cval));
+    cache->eviction_dirty_target = (double)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_dirty_trigger", &cval));
+    cache->eviction_dirty_trigger = (double)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_updates_target", &cval));
+    cache->eviction_updates_target = (double)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_updates_trigger", &cval));
+    cache->eviction_updates_trigger = (double)cval.val;
+
+    WT_RET(__wt_config_gets(session, cfg, "eviction_checkpoint_target", &cval));
+    cache->eviction_checkpoint_target = (double)cval.val;
+
+    /* Check for invalid configurations and automatically fix them to suitable values. */
+    if (cache->eviction_dirty_target > cache->eviction_target) {
+        if (debug_config)
+            __wt_verbose_warning(session, WT_VERB_CONFIGURATION,
+              "config eviction_dirty_target=%f is larger than eviction_target=%f, changing "
```

Comment:
General wording suggestion for this and the other warning messages
```suggestion
              "config eviction_dirty_target (%f) cannot exceed than eviction_target (%f). Setting"
```

Reply:
Comment rephrased in: [9e4baa7](https://github.com/wiredtiger/wiredtiger/pull/10548/commits/9e4baa7696ed61b4507e4766121155606277fcd8)